### PR TITLE
Add DiffMerge tool support

### PR DESCRIPTION
**Pull request by akantsevoi**
Close the issue https://github.com/git-up/GitUp/issues/59

### CODE REVIEW

**Comment by lucasderraugh** [comment]
@akantsevoi I've recently taken over as temporary maintainer of GitUp. If I can get an overview of visual changes, I'd love to start the discussion on merging this in.

**Comment by akantsevoi** [comment]
Hi @lucasderraugh , nice to meet you. What do you mean about visual changes? Some screenshots or like that?

**Comment by lucasderraugh** [comment]
@akantsevoi Nice to meet you too! Screenshots of any visual changes would be great! Along with just a brief explanation of the code and what you're trying to accomplish with it. Just helps when I'm testing out the change to know what differences I should be looking for.

File: GitUpKit/Utilities/GIViewController+Utilities.m [hunk 1]
```diff
@@ -571,6 +583,15 @@ - (void)resolveConflictInMergeTool:(GCIndexConflict*)conflict {
...
     [self _runMergeGitToolForFile:mergePath withOldPath:ourPath newPath:theirPath basePath:ancestorPath];
+  } else if ([identifier isEqualToString:GIViewControllerTool_DiffMerge]) {
+    [arguments addObject:[NSString stringWithFormat:@"-r=%@", mergePath]];
+    [arguments addObject:[NSString stringWithFormat:@"-t1=%@", ourTitle]];
+    [arguments addObject:[NSString stringWithFormat:@"-t2=%@", ancestorTitle]];
+    [arguments addObject:[NSString stringWithFormat:@"-t3=%@", theirTitle]];
+    [arguments addObject: ourPath];
+    [arguments addObject: ancestorPath];
+    [arguments addObject: theirPath];
+    [self _runDiffMergeToolWithArguments: arguments];
```
Comment (lucasderraugh) [inline GitUpKit/Utilities/GIViewController+Utilities.m]: Could we remove the spacing after the colon in these 4 spots. Other than that, code wise this looks good to me.

**Comment by swisspol** [comment]
You can just run format-source script after merging

**Comment by lucasderraugh** [comment]
@swisspol Do you just run that periodically and make a new commit? Would be nice if that could somehow be done before merging, but it's good to know that script exists.

**Comment by swisspol** [comment]
I just ran it by hand before committing or sometimes after PRs

**Comment by lucasderraugh** [comment]
@akantsevoi How would I go about testing out this change? Some steps to verify would be great so I can push this change along. Thanks!

**Comment by lucasderraugh** [comment]
Somehow there is a merge commit that I didn't want but I guess GitHub Desktop insisted on and so it's screwed this branch. Definitely don't want to make the changes that I've put on here now, but I'll take what you had and create a new PR. Sorry for this, was just trying to resolve a xib file conflict...

**Comment by lucasderraugh** [comment]
Sorry @akantsevoi. I've rebased your changes on master and they're set to be taken whenever I can fix the continuous build signing issues. Your changes were added with 6aa8df8b98117ebe8eb3cbbe26f991e4a71cfda6

**Pull request closed without merging**